# EXPLORE HYPOTHESIS TESTING
## Introduction
You work for an environmental think tank called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health.

They've tasked you with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, construct a hypothesis test and an accompanying visualization, using your results of that test to make a recommendati1. on:

ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of Calif2. ornia.
With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI tha3. n Ohio?
A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new
##  policy?
Notes:

For your analysis, you'll default to a 5% level of significance.
Throughout the lab, for two-sample t-tests, use Welch's t-test (i.e., setting the equal_var parameter to False in scipy.stats.ttest_ind()). This will account for the possibly unequal variances between the two groups in the comparison

In [3]:
import pandas as pd
import numpy as np
from scipy import stats

In [4]:
filepath= r"C:\Users\chile\OneDrive\Desktop\MIT Lecture Notes\PYTHON DATA\c4_epa_air_quality.csv"

In [5]:
aqi= pd.read_csv(filepath)

In [6]:
aqi.head()

,Unnamed: 0,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
0,0,2018-01-01,Arizona,Maricopa,Buckeye,BUCKEYE,Carbon monoxide,Parts per million,0.473684,7
1,1,2018-01-01,Ohio,Belmont,Shadyside,Shadyside,Carbon monoxide,Parts per million,0.263158,5
2,2,2018-01-01,Wyoming,Teton,Not in a city,Yellowstone National Park - Old Faithful Snow ...,Carbon monoxide,Parts per million,0.111111,2
3,3,2018-01-01,Pennsylvania,Philadelphia,Philadelphia,North East Waste (NEW),Carbon monoxide,Parts per million,0.300000,3
4,4,2018-01-01,Iowa,Polk,Des Moines,CARPENTER,Carbon monoxide,Parts per million,0.215789,3


In [6]:
print(aqi.head())
print(aqi.describe(include= 'all'))
print(aqi['state_name'].value_counts())


   Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158 

### Step 3. Statistical Tests
Before you proceed, recall the following steps for conducting hypothesis testing:


Formulate the null hypothesis and the alternative hypothes
i1. s.
Set the significance le
2. 
3. el.
Determine the appropriate test proce
4. ure.
Compute the p-
5. alue.
Draw your conclusion.

## Hypothesis 1: 
ROA is considering a metropolitan-focused approach. Within California, 
they want to know if the mean AQI in Los Angeles County is statistically
different from the rest of California.

In [9]:
ca_la = aqi[aqi['county_name']=='Los Angeles']
ca_other = aqi[(aqi['state_name']=='California') & (aqi['county_name']!='Los Angeles')]

 ## Formulate your null and alternative hypotheses:

𝐻0
 : There is no difference in the mean AQI between Los Angeles County and the rest of California.

𝐻𝐴

 : There is a difference in the mean AQI between Los Angeles County and the rest of California.

In [10]:
significane_level= 0.05

## Calculate P-value

In [11]:
stats.ttest_ind(a= ca_la['aqi'], b= ca_other['aqi'], equal_var= False)

TtestResult(statistic=2.1107010796372014, pvalue=0.04983905684241103, df=17.08246830361151)

We reject the Ho as the p-value is less than 0.05

## Hypothesis 2: 
With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [7]:
# Create dataframes for each sample being compared in your test

### YOUR CODE HERE ###
NY = aqi[aqi['state_name']=='New York']
Ohio= aqi[aqi['state_name']=='Ohio']


ormulate your null and alternative hypotheses:

𝐻0
 : The mean AQI of New York is greater than or equal to that of Ohio.
𝐻𝐴
 : The mean AQI of New York is below that of Ohio.

Here, you are comparing the sample means between two independent samples in one direction. Therefore, you will utilize a two-sample 𝑡-test.

Compute the p-value

In [8]:
tstat, pvalue= stats.ttest_ind(a=NY['aqi'], b= Ohio['aqi'], alternative= 'less', equal_var= False )

In [15]:
print(tstat)
print(pvalue)

-2.025951038880333
0.030446502691934687


In [12]:
# Or you can use this
stats.ttest_ind(a=NY['aqi'], b= Ohio['aqi'], alternative= 'less', equal_var= False)

TtestResult(statistic=-2.025951038880333, pvalue=0.030446502691934687, df=15.036745051598716)

With a p-value (0.030) of less than 0.05 (as your significance level is 5%) and a t-statitics < 0 (-2.036), reject the null hypothesis in favor of the alternative hypothesis.

Therefore, you can conclude at the 5% significance level that New York has a lower mean AQI than Ohio.

## Hypothesis 3:
A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?
Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [16]:
# Create dataframes for each sample being compared in your test

### YOUR CODE HERE ###

michigan = aqi[aqi['state_name']=='Michigan']

Formulate your null and alternative hypotheses here:

𝐻0
 : The mean AQI of Michigan is less than or equal to 10.
𝐻𝐴
 : The mean AQI of Michigan is greater than 10.

 sig_level= 0.05

Here, you are comparing one sample mean relative to a particular value in one direction. Therefore, you will utilize a one-sample 𝑡-test.

Compute the P-valuem

In [18]:
# Compute P-Value
tstat, pvalue= stats.ttest_1samp(michigan['aqi'], 10, alternative= 'greater')
print(tstat)
print(pvalue)

-1.7395913343286131
0.9399405193140109


## Reject or Accept
With a p-value (0.940) being greater than 0.05 (as your significance level is 5%) and a t-statistic < 0 (-1.74), fail to reject the null hypothesis.

Therefore, you cannot conclude at the 5% significance level that Michigan's mean AQI is greater than 10. This implies that Michigan would most likely not be affected by the new policy.

# Conclusion

**What are key takeaways from this lab?**

Even with small sample sizes, the variation within the data is enough to allow you to make statistically significant conclusions. You identified at the 5% significance level that the Los Angeles mean AQI was stastitically different from the rest of California, and that New York does have a lower mean AQI than Ohio. However, you were unable to conclude at the 5% significance level that Michigan's mean AQI was greater than 10.

**What would you consider presenting to your manager as part of your findings?**

For each test, you would present the null and alternative hypothesis, then describe your conclusion and the resulting p-value that drove that conclusion. As the setup of t-test's have a few key configurations that dictate how you interpret the result, you would specify the type of test you chose, whether that tail was one-tail or two-tailed, and how you performed the t-test from `stats`.


**What would you convey to external stakeholders?**

In answer to the research questions posed, you would convey the level of significance (5%) and your conclusion. Additionally, providing the sample statistics being compared in each case will likely provide important context for stakeholders to quickly understand the difference between your results.
